### ISSUE TITLE
[Operator] Blocking issue : Please don't add ownerReferences to Vault-tls secret

### ISSUE THREAD

**Issue by sparqueur** [banzaicloud/bank-vaults#899]
We have currently a blocking issue when restarting our K8S cluster.

All objects "linked" (ownerReferences) to the Vault CRD are destroyed by Kubernetes on restart (not relative to you).

Problem : some other deployment may already be using the Vault-tls secret! => Error for them on startup

In the incoming version you already removed the ownerReferences from the unseal keys secret but this is not enough, we need this same behavior on vault-tls secret which is managed by the operator

You may add an extra argument to parametrized this behavior.

Moreover I figured out that when I remove the ownerReferences from the vault-tls secret, kubernetes does not destroy our vault instance, perfect !
Unfortunatly the operator always add ownerReferences back so I have to first stop it.

Thank's in advance

**Comment by bonifaido** [comment]
Hi @sparqueur,

Yes, we have removed setting the ownerReference of the `vault-unseal-key` Secret in https://github.com/banzaicloud/bank-vaults/pull/866 but mostly because of operational concerns, so the Vault data and the corresponding unseal keys are having the same life period. That PR might fix some other cluster restart/object disappearance issues, but only as a side effect, at least not intentionally. :)

> All objects "linked" (ownerReferences) to the Vault CRD are destroyed by Kubernetes on restart (not relative to you).

Why does this happen in your cluster? Probably this issue should be resolved rather in my opinion.

**Comment by sparqueur** [comment]
Hi,
I obviously agree with you that it should be fixed by Kubernetes.
We currently use an RKE cluster, it might be relative to it.

Unfortunatly for us, this problem might be a no go for using bank vault in production as we are not abble to restart our cluster without having some startup issues :(

May it be possible to you to set the ownerReferences only when creating the object ?
I am able to remove it using some scripts but it always comes back.

I will try to find out what's happening in our cluster

**Comment by primeroz** [comment]
> as we are not abble to restart our cluster without having some startup issues :(

I am assuming this is a rolling restart ? 
Just to add to the point, i run 3 production clusters with the vault operator running in GKE , each of them i upgraded 3 times with rolling master restarts and never had an issue

That's why i think that removing the `ownerReference` would just be a workaround rather than a fix of the root issue 

> May it be possible to you to set the ownerReferences only when creating the object ?

Have you tried https://github.com/banzaicloud/bank-vaults/pull/859 ? 
With that you are able to completely disable the `vault-tls` secret management from the operator and will use your own secret, that should totally be `ownerReference` free 

Not sure if is released yet but is definetely in master

**Comment by sparqueur** [comment]
Hi @primeroz,
It is not a normal scenario I am simulating, more a disaster recovery plan when server are restating.
I may have to manage certificates, it's just I was hopping I could rely on automatic certificate management (with automatic renewal ...).
I've just recorded the crazy behavior : https://www.youtube.com/watch?v=Jnvj0ylhBqU&feature=youtu.be
Note that I first stopped vault operator and webhook (so I can be sure they are not responsible of it).
Moreover, if I turn down vault instance and configurer to 0 then they are not always deleted / recreated :
- If I only have vault in the cluster => OK
- If I have other stuff running (even terminated jobs) => They are deleted
I m getting crazy :-)
